mediacapture-streams/MediaDevices-enumerateDevices-per-origin-ids.sub.https.html

```diff
@@ -26,7 +26,8 @@
     const iframeDevices = e.data.devices;
     assert_equals(deviceList.length, iframeDevices.length, "Same number of devices detected same-origin");
     for (const device of deviceList) {
-      const sameDevice = iframeDevices.find(d => d.deviceId === device.deviceId);
+      // special casing "default" deviceId
+      const sameDevice = iframeDevices.find(d => d.deviceId === device.deviceId && (device.deviceId !== "default" || d.kind === device.kind));
```

Comment:
this means that (kind=audio, id=default) does not match (kind=video, id=default), but it is a bit hard to see that that's what it's computing.
Under what circumstance would the "deviceId !== default" make a difference? two devices should never have the same id unless it's "default".

Reply:
This allows to test on L33 that a device with the same `deviceId` is of the same kind; `default` is treated as an exception (since it can be reused across kinds).

IOW, the test on L33 is useless for `default` devices, but meaningful for all other devices.

I've updated the comment to clarify the computation and its intent